From fd03583ceef172c6a97b53e99876ca57c478ffaf Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:01:46 -0400
Subject: [usb] anchor: Ensure poisened gets initialized to 0

Message-id: <1410887463-20674-35-git-send-email-dzickus@redhat.com>
Patchwork-id: 95114
O-Subject: [RHEL7 PATCH 034/591] usb-anchor: Ensure poisened gets initialized to 0
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 9ef73dbdd0fc292d183e93cd1d4b21d1a66040d7
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Wed Oct 9 17:01:40 2013 +0200

    usb-anchor: Ensure poisened gets initialized to 0

    And do so in a way which ensures that any fields added in the future will
    also get properly zero-ed.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Acked-by: Oliver Neukum <oliver@neukum.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/usb.h b/include/linux/usb.h
index d8e0b10..dd7904c 100644
--- a/include/linux/usb.h
+++ b/include/linux/usb.h
@@ -1214,6 +1214,7 @@ struct usb_anchor {
 
 static inline void init_usb_anchor(struct usb_anchor *anchor)
 {
+ memset(anchor, 0, sizeof(*anchor));
  INIT_LIST_HEAD(&anchor->urb_list);
  init_waitqueue_head(&anchor->wait);
  spin_lock_init(&anchor->lock);
-- 
1.7.1